File: mobile/android/VehicleDiagnosticModel.java
package com.autoconnect.android;

/** A single diagnostic trouble code (DTC) result. */
public class VehicleDiagnosticModel {
    public String code;
    public String description;
    public String system;
    public String severity; // HIGH | MEDIUM | LOW | UNKNOWN
    public String recommendedAction;

    public VehicleDiagnosticModel(String code, String description, String system,
                                  String severity, String recommendedAction) {
        this.code = code;
        this.description = description;
        this.system = system;
        this.severity = severity;
        this.recommendedAction = recommendedAction;
    }

    public boolean isCritical() {
        return "HIGH".equalsIgnoreCase(severity);
    }
}

File: mobile/ios/VehicleDiagnosticModel.swift
import Foundation

/// A single diagnostic trouble code (DTC) result.
struct VehicleDiagnosticModel: Codable, Identifiable {
    var id: String { code }
    let code: String
    let description: String
    let system: String
    let severity: String  // HIGH | MEDIUM | LOW | UNKNOWN
    let recommendedAction: String

    enum CodingKeys: String, CodingKey {
        case code, description, system, severity
        case recommendedAction = "recommended_action"
    }

    var isCritical: Bool { severity.uppercased() == "HIGH" }
}
